core/src/io/anuke/mindustry/core/NetServer.java

```diff
@@ -74,8 +74,9 @@ public NetServer(){
         });
 
         Net.handleServer(Disconnect.class, (id, packet) -> {
-            Player player = connections.get(id);
+            Player player = connections.get(id);           
             if(player != null){
+                Events.fire(new PlayerLeave(player)); 
```

Comment:
This won't accurately fire for all player leaving events. Instead, fire the event in `onDisconnect(...)` inside the `hasConnected` if-statement, before the message is sent.